intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+# An introduction to calculating Cred and Grain
+
+I’m starting from the assumption that you have some familiarity with the SourceCred project. If you don’t I’d suggest you check out Dandelion's [20 minute talk (+Qs) from SustainWeb3](https://www.youtube.com/watch?v=yVTqRLekRl4), and some of the other documents in this repo.
+
+Overall, I’m going to throw around some math, but you should be able to get solid intuition on this if you’re comfortable talking about a [graph](https://en.wikipedia.org/wiki/Graph_theory) as a set of nodes and edges connecting them, where the edges can be directed and have weights. I’ll also talk some about [the PageRank algorithm](https://en.wikipedia.org/wiki/PageRank), for computing “importance” of nodes in a graph based on the “importance” of nodes connected to it, but I’ll try to motivate the intuition behind it as well.
+
+## The high level overview
+
+SourceCred takes information about contributions and generates a weights graph, and this post will first explain what that graph is and where it comes from. From this weights graph, it runs PageRank to generate a graph of Cred. This graph of Cred is then used to distribute grain. The grain can then be kept or traded, and soon will be able to be used to influence future graph weights. This process is diagrammed in the figure below.
```

Comment:
weights graph or weighted graph? (I've seen both around!)

Reply:
Does a link to PageRank make sense? Or maybe a reader friendly extra sentence like "Don't know what PageRank is? That's okay, because we'll cover this [here]() / that's okay, you can learn more [here]()" etc.

Reply:
Yeah, I'm growing increasingly confident that I can explain PageRank and stable distributions without heavy math, and I'd much rather do that than link away to an explanation elsewhere.  I'll comment on this thread again once I've made those edits.

Reply:
Also, I should definitely make a distinction between a weighted graph (a type of graph with numbers on the edges) and The Weights Graph, the SourceCred graph of contributions with numbers associated with edges and nodes.  Having read your comment and thought about this more, I definitely want to make the distinction clearer, and will comment here once I've made those changes.

Reply:
We generally call it a "contribution graph". Calling it a "weights" graph is confusing IMO: what's most interesting about the graph is that it represents contributions and connections between them; the fact that we also store weights in the graph is not as conceptually important.

Reply:
Contribution conveys better the functionality vs weight that refers to the method of implementation.

Reply:
It's definitely now "contribution graph" throughout the doc.  👍